solr/solrj/src/java/org/apache/solr/common/util/Utils.java

```diff
@@ -302,10 +302,16 @@ public static byte[] toUTF8(CharArr out) {
   }
 
   public static Object fromJSON(byte[] utf8) {
+    if (utf8 == null || utf8.length == 0) {
+      return Collections.emptyMap();
+    }
     return fromJSON(utf8, 0, utf8.length);
   }
 
   public static Object fromJSON(byte[] utf8, int offset, int length) {
+    if (utf8 == null || utf8.length == 0) {
```

Comment:
Check length parameter here too?

Reply:
Fixed in 2011fb0c51476478322e41a4f2b5259f44ea6662